Dangerfile

```diff
@@ -13,16 +13,18 @@ avoid_exact_words = [
   { word: 'react native', reason: 'Please use React Native with capitals' }
 ]
 
-active_files = (git.modified_files + git.added_files)
+active_files = (git.modified_files + git.added_files).uniq
 markdowns = active_files.select { |file| file.start_with? '_posts/' }
 
 # This could do with some code golfing sometime
-markdowns.each do |file|
-  lines = File.read(file).lines
+markdowns.each do |filename|
+  file = File.read(filename)
+  lines = file.lines
+  fail("Please add a <!-- more --> tag where you'd like this post to break for post preview", file: filename) unless file.include?("<!-- more -->")
```

Comment:
Nice!

One thought I have is to use `File.open` and `File#none?` so that we that we:
1. Don't have to load the entire file content into memory
2. Can quickly exit the check if the 'more' tag is found early in the file

```ruby
markdowns.each do |filename|
  File.open("filename", "r) do |file| # file is a File object here
    if file.none? { |line| line.include?("<!-- more -->") }
      fail("Please add a <!-- more --> tag where you'd like this post to break for post preview", file: filename)
    end
    file.rewind # resets the reader so that the next `#each` block iterates over all lines

    file.each do |line|
      avoid_exact_words.each do |avoid|
        warn(avoid[:reason], file: filename, line: line) if line.include? avoid[:word]
       end
    end
  end
end
```

Although, our set of active files will likely be small and the files we're scanning are unlikely to be large enough for this to make difference.

Reply:
@dblandin The `fail` use here is a little weird – this won't be using Ruby's `fail`, but Danger's (it's since been renamed to `failure` to disambiguate, but our install is outdated). So no matter what, the entire loop will be executed; a post missing the `more` tag and with an `avoid_exact_word` will get both a failure and a warning. The additional benefit of loading the whole file into memory is that we can repeatedly access its contents for other Danger rules without having to rewind. 

I know Ruby has a bunch of different ways to handle files – are there others we could implement with the context I mentioned?

Reply:
Ahh! Thanks for that context! In that case, we might consider restructuring the loop so that we're only iterating over a file's lines once. No need to rewind as long as all of your rules are checking lines individually.

```ruby
markdowns.each do |filename|
  File.open(filename, "r") do |file|
    file.lines.each do |line|
      # Assert presence of <!-- more --> tag
      if line.include?("<!-- more -->")
        fail("Please add a <!-- more --> tag where you'd like this post to break for post preview", file: filename)
      end

      # Discourage use of words/spellings
      avoid_exact_words.each do |avoid|
        warn(avoid[:reason], file: filename, line: line) if line.include? avoid[:word]
       end
    end
  end
end
```

Reply:
```rb
if line.include?("<!-- more -->")
  fail("Please add a <!-- more --> tag where you'd like this post to break for post preview", file: filename)
end
```

Wouldn't this check to make sure there are _no_ more tags? I couldn't think of a way to verify the presence of a particular string in the file without having it all loaded (without doing some hacky `found_more = false` on the outside of the loop).

Reply:
Ahhh, yes, you're right. That snippet didn't translate well between those two examples. Pretty sure what you already have in this PR is the best option. :+1: